falcon/util/uri.py

```diff
@@ -309,7 +317,7 @@ def parse_query_string(query_string, keep_blank_qs_values=False):
                 params[k] = [old_value, decode(v)]
 
         else:
-            if ',' in v:
+            if ',' in v and csv:
```

Comment:
it's in the wrong order, `',' in v` is evaluated before `csv`. If it should short-circuit it has to be `csv and ',' in v`.